tests/xharness/TestReporter.cs

```diff
@@ -0,0 +1,490 @@
+					(parseResult.resultLine, parseResult.failed) = ResultParser.GenerateHumanReadableResults (path, tmpFile, xmlType);
+					File.Copy (tmpFile, test_log_path, true);
+					File.Delete (tmpFile);
+
+					// we do not longer need the tmp file
+					runner.Logs.AddFile (path, LogType.XmlLog.ToString ());
+					return parseResult;
+
+				} catch (Exception e) {
+					runner.MainLog.WriteLine ("Could not parse xml result file: {0}", e);
+					// print file for better debugging
+					runner.MainLog.WriteLine ("File data is:");
+					runner.MainLog.WriteLine (new string ('#', 10));
+					using (var stream = new StreamReader (path)) {
+						string line;
+						while ((line = await stream.ReadLineAsync ()) != null) {
+							runner.MainLog.WriteLine (line);
+						}
+					}
+					runner.MainLog.WriteLine (new string ('#', 10));
+					runner.MainLog.WriteLine ("End of xml results.");
+					if (timed_out) {
+						WrenchLog.WriteLine ($"AddSummary: <b><i>{runner.RunMode} timed out</i></b><br/>");
+						return parseResult;
+					} else {
+						WrenchLog.WriteLine ($"AddSummary: <b><i>{runner.RunMode} crashed</i></b><br/>");
+						runner.MainLog.WriteLine ("Test run crashed");
+						parseResult.crashed = true;
+						return parseResult;
+					}
+				}
+
+			}
+			// delete not needed copy
+			File.Delete (path);
+
+			// not the most efficient way but this just happens when we run
+			// the tests locally and we usually do not run all tests, we are
+			// more interested to be efficent on the bots
+			(parseResult.resultLine, parseResult.failed) = await GetResultLine (test_log_path);
+			return parseResult;
+		}
+
+		async Task<(bool Succeeded, bool Crashed)> TestsSucceeded (AppBundleInformation appInfo, string test_log_path, bool timed_out)
+		{
+			var (resultLine, failed, crashed) = await ParseResultFile (appInfo, test_log_path, timed_out);
+			// read the parsed logs in a human readable way
+			if (resultLine != null) {
+				var tests_run = resultLine.Replace ("Tests run: ", "");
+				if (failed) {
+					WrenchLog.WriteLine ("AddSummary: <b>{0} failed: {1}</b><br/>", runner.RunMode, tests_run);
+					runner.MainLog.WriteLine ("Test run failed");
+					return (false, crashed);
+				} else {
+					WrenchLog.WriteLine ("AddSummary: {0} succeeded: {1}<br/>", runner.RunMode, tests_run);
+					runner.MainLog.WriteLine ("Test run succeeded");
+					return (true, crashed);
+				}
+			} else if (timed_out) {
+				WrenchLog.WriteLine ("AddSummary: <b><i>{0} timed out</i></b><br/>", runner.RunMode);
+				return (false, false);
+			} else {
+				WrenchLog.WriteLine ("AddSummary: <b><i>{0} crashed</i></b><br/>", runner.RunMode);
+				runner.MainLog.WriteLine ("Test run crashed");
+				return (false, true);
+			}
+		}
+
+		// generate all the xml failures that will help the integration with the CI and return the failure reason
+		async Task GenerateXmlFailures (string failureMessage, bool crashed, string crashReason)
+		{
```

Comment:
Big change, we do not longer create a crash xml failure per crash report. we only do it when we crash.